lms/djangoapps/grades/tests/test_signals.py

```diff
@@ -148,3 +160,95 @@ def test_score_reset_bad_user(self):
         submissions_score_reset_handler(None, **SUBMISSION_RESET_KWARGS)
         self.signal_mock.assert_not_called()
+
+
+@ddt.ddt
+class ScoreChangedUpdatesSubsectionGradeTest(ModuleStoreTestCase):
+    """
+    Ensures that upon SCORE_CHANGED signals, the handler
+    initiates an update to the affected subsection grade.
+    """
+    def setUp(self):
+        super(ScoreChangedUpdatesSubsectionGradeTest, self).setUp()
+        self.user = UserFactory()
+
+    def set_up_course_and_user(self, enable_subsection_grades):
+        """
+        Configures the course for this test.
+        """
+        # pylint: disable=attribute-defined-outside-init,no-member
+        self.course = CourseFactory.create(
+            org='edx',
+            name='course',
+            run='run',
+            metadata={'enable_subsection_grades_saved': enable_subsection_grades})
+
+        self.chapter = ItemFactory.create(parent=self.course, category="chapter", display_name="Chapter")
+        self.sequential = ItemFactory.create(parent=self.chapter, category='sequential', display_name="Open Sequential")
+        self.problem = ItemFactory.create(parent=self.sequential, category='problem', display_name='problem')
+
+        self.score_changed_kwargs = {
+            'points_possible': 10,
+            'points_earned': 5,
+            'user': self.user,
+            'course_id': unicode(self.course.id),
+            'usage_id': unicode(self.problem.location),
+        }
+
+        # this call caches the anonymous id on the user object, saving 4 queries in all happy path tests
+        _ = anonymous_id_for_user(self.user, self.course.id)
+        # pylint: enable=attribute-defined-outside-init,no-member
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_subsection_grade_updated_on_signal(self, default_store):
+        with self.store.default_store(default_store):
+            self.set_up_course_and_user(True)
+            with check_mongo_calls(2) and self.assertNumQueries(13):
+                recalculate_subsection_grade_handler(None, **self.score_changed_kwargs)
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_query_count_does_not_change_with_more_problems(self, default_store):
+        with self.store.default_store(default_store):
+            self.set_up_course_and_user(True)
+            ItemFactory.create(parent=self.sequential, category='problem', display_name='problem2')
+            ItemFactory.create(parent=self.sequential, category='problem', display_name='problem3')
+            with check_mongo_calls(2) and self.assertNumQueries(13):
+                recalculate_subsection_grade_handler(None, **self.score_changed_kwargs)
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_subsection_grades_not_enabled_on_course(self, default_store):
+        with self.store.default_store(default_store):
+            self.set_up_course_and_user(False)
```

Comment:
Nit: for code readability, it's generally a good idea to specify the argument name when passing booleans.  So here, `False` can be replaced with `enable_subsection_grades=False`.